### PR TITLE
Disable pprint line wrapping in the doctests

### LINKED ISSUE #26318: Doctest failures due to line wrapping in the printers
Realted to gh-26221.

After gh-25673 the printers are changed to add extra characters when wrapping lines. This causes the doctests to fail when run outside of CI e.g.:
```console
$ bin/doctest sympy/categories/baseclasses.py
================================================================================================ test process starts =================================================================================================
executable:         /Users/enojb/.pyenv/versions/sympy-3.12.git/bin/python  (3.12.2-final-0) [CPython]
architecture:       64-bit
cache:              yes
ground types:       python
numpy:              1.26.4
hash randomization: on (PYTHONHASHSEED=3926478540)

sympy/categories/baseclasses.py[23] .......F...............                                                                                                                                                     [FAIL]

______________________________________________________________________________________________________________________________________________________________________________________________________________________
________________________________________________________________________________________ sympy.categories.baseclasses.Diagram ________________________________________________________________________________________
File "/Users/enojb/work/dev/sympy/sympy/categories/baseclasses.py", line 611, in sympy.categories.baseclasses.Diagram
Failed example:
    pprint(d.premises, use_unicode=False)
Expected:
    {g*f:A-->C: EmptySet, id:A-->A: EmptySet, id:B-->B: EmptySet, id:C-->C: EmptyS >
    <BLANKLINE>
    > et, f:A-->B: EmptySet, g:B-->C: EmptySet}
Got:
    {g*f:A-->C: EmptySet, id:A-->A: EmptySet, id:B-->B: EmptySet, id:C-->C: EmptySet, f:A-->B: EmptySet, g:B-->C: EmptySet}

================================================================================ tests finished: 22 passed, 1 failed, in 0.12 seconds ================================================================================
DO *NOT* COMMIT!
```
The problem seems to be that the doctests effectively hardcode a particular terminal width that works in CI but not anywhere else.

I think that this printing change should be reverted but either way the doctests should be made to work.

CC @smichr

### PR DESCRIPTION
**Pull request by asmeurer**
This makes the doctests pass locally, as were currently hard-coded to whatever
column length was set on CI.

Note that single line outputs can still be manually line wrapped, as the
doctester ignores newlines. Multiline outputs will need to either be included
as the full length or manually wrapped by passing num_columns to pprint() (or
something other than pprint used).

Fixes #26318.

#### Release Notes

NO ENTRY

### CODE REVIEW

**Comment by asmeurer** [comment]
I couldn't find anywhere that was setting `$COLUMNS` on CI. It must have just been using some default value. At any rate, I confirmed locally with my wide terminal window that the doctests now pass.

**Comment by asmeurer** [comment]
Obviously there's a lot of related cleanups that could be made here. Many of these instances probably could just demonstrate the expression using inline LaTeX. Also I don't know if we should be using Unicode pprint anywhere but we apparently are (the formatting of those pages will look bad on computers without proper font support, see #15700). Also the continuation stuff itself is still up for discussion #26221. 

But we can fix these issues separately. My goal here was just to fix the doctest issues that were blocking the release.

**Comment by oscarbenjamin** [comment]
> I couldn't find anywhere that was setting `$COLUMNS` on CI. It must have just been using some default value

Is it this:
https://github.com/sympy/sympy/blob/ae3027b4ca682023dee36e7e8883311c62cc08e3/sympy/printing/pretty/stringpict.py#L320-L324

That was recently modified in 992eaf56523 from gh-25753.

**Comment by asmeurer** [comment]
Looks like doctests are failing on CI, though, so I'll need to figure that out.

**Comment by asmeurer** [comment]
Perhaps I was being fooled by my terminal already being extra wide and my fix doesn't actually work.

**Comment by oscarbenjamin** [comment]
> I just tried locally with a wide and then a narrow terminal and it seemed to work (with this PR).

I only tested the docstrings. For the rst docs it fails with pytest-doctestplus and also with bin/doctest.

**Comment by asmeurer** [comment]
I'm a little confused why this test passes on master, and why the changes here make it fail

```py
    huge_expr = 0
    for i in range(20):
        huge_expr += i*sin(i + x)
    assert xpretty(huge_expr            ).find('\n') != -1
```

If I run `pretty(huge_expr)` interactively, the default is to line wrap, so it does have newlines.

**Comment by asmeurer** [comment]
Not sure what to do about pytest-doctestplus. In principle, it should just need to call `setup_pprint` similar to how the normal doctester does.

**Comment by oscarbenjamin** [comment]
> Not sure what to do about pytest-doctestplus

We don't need to worry about pytest-doctestplus right now.

**Comment by asmeurer** [comment]
I figured out the tests. 

The code here turning off and on global settings for the tests is a little messy. Any suggestions for improving it are welcome. We should probably unify the line wrapping and the unicode logic somehow.

File: conftest.py [hunk 1]
```diff
@@ -16,7 +16,7 @@
 collect_ignore = _get_doctest_blacklist()
 
 # Set up printing for doctests
-setup_pprint()
+setup_pprint(disable_line_wrap=False)
```
Comment (asmeurer) [inline conftest.py:19]: Presumably pytest-doctestplus would actually do the right thing without this flag, but this also breaks the line pprint tests that test the line wrapping defaults.

**Pull request merged**